sycl/src/utils/sycl/opaque_queue.hpp

```diff
@@ -0,0 +1,20 @@
+/** VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+// SYCL include(s).
+#include <CL/sycl.hpp>
+
+namespace vecmem::sycl::details {
+
+   /// Helper class for managing queue objects in memory
+   class opaque_queue : public cl::sycl::queue {
```

Comment:
What is the motivation for this class?

Reply:
Ha... This is something that I took from https://github.com/root-project/root/commit/289c1d701d8d36cecd8376f03ffc9651644dd315. :smile:

Intel (as we see in TBB's example) likes to define its types in fairly complicated ways. At the moment `cl::sycl::queue` is a "real class". But at any point it can become a typedef. This sort of design is meant to guard us against that. While still making it possible to use `std::unique_ptr` for managing the queue's memory.